### PR TITLE
Fix Crash EditDescriptionActivity when switched dark/light mode

### LINKED ISSUE #5473: [Bug]: Changes Lost and App Crashes when Switched between Dark-Light Modes in Edit Description and Captions
### Summary

In Edit Description and Caption Page when you switch between Dark-Light Modes then the captions are lost and app eventually crashes.

### Steps to reproduce

1. Open the App
2. Go to Any Contribution
3. Click on the Edit Icon For the Description 
4. Switch Dark Mode/Light Mode
5. Press the Back Button or Submit Button

### Expected behaviour

Edited captions and descriptions should be retained

### Actual behaviour

Changes are Lost and the App Crashes

### Device name

Xaomi 11 Lite NE

### Android version

13

### Commons app version

4.2.1-debug-main

### Device logs

_No response_

### Screen-shots

https://github.com/commons-app/apps-android-commons/assets/126143257/dd6c8d9c-b8ee-471c-b211-eba587598f4d

### Would you like to work on the issue?

Yes

### PR DESCRIPTION
**Pull request by shashankiitbhu**
**Description (required)**

Fixes #5473 

What changes did you make and why?

**Tests performed (required)**

Tested {build variant, e.g. ProdDebug} on {name of device or emulator} with API level {API level}.

**Screenshots (for UI changes only)**

### CODE REVIEW

**Comment by rohit9625** [comment]
Please mention the issue number that this PR solves : )

**Comment by shashankiitbhu** [comment]
Yeah I know I was going to mention the issue number once a related PR is merged, I want to make a commit after that PR is approved , so I didn't want  to get this one reviewed before that :)

**Comment by rohit9625** [comment]
Ohh I see. 😅

File: app/src/main/java/fr/free/nrw/commons/description/DescriptionEditActivity.kt [hunk 1]
```diff
@@ -60,17 +67,30 @@ class DescriptionEditActivity : BaseActivity(), UploadMediaDetailAdapter.EventLi
...
         binding = ActivityDescriptionEditBinding.inflate(layoutInflater)
         setContentView(binding.root)
 
         val bundle = intent.extras
-        val descriptionAndCaptions: ArrayList<UploadMediaDetail> =
-            bundle!!.getParcelableArrayList(LIST_OF_DESCRIPTION_AND_CAPTION)!!
-        wikiText = bundle.getString(WIKITEXT)
-        savedLanguageValue = bundle.getString(Prefs.DESCRIPTION_LANGUAGE)!!
+
+        if(savedInstanceState!=null){
```
Comment (RitikaPahwa4444) [inline app/src/main/java/fr/free/nrw/commons/description/DescriptionEditActivity.kt:82]: Need some time to test on a device, but I skimmed through the code. Just a few suggestions based on the [style guide](https://github.com/commons-app/commons-app-documentation/blob/master/android/Code-style.md) the app follows: 
- space after `if`
- space before and after `!=`
- space after `)`

File: app/src/main/java/fr/free/nrw/commons/description/DescriptionEditActivity.kt [hunk 2]
```diff
@@ -60,17 +67,30 @@ class DescriptionEditActivity : BaseActivity(), UploadMediaDetailAdapter.EventLi
...
+            descriptionAndCaptions = savedInstanceState.getParcelableArrayList(LIST_OF_DESCRIPTION_AND_CAPTION)
+            wikiText = savedInstanceState.getString(WIKITEXT)
+            savedLanguageValue = savedInstanceState.getString(Prefs.DESCRIPTION_LANGUAGE)!!
+            media = savedInstanceState.getParcelable("media")
+        }else{
```
Comment (RitikaPahwa4444) [inline app/src/main/java/fr/free/nrw/commons/description/DescriptionEditActivity.kt:87]: For this as well: space before and after `else`

File: app/src/test/kotlin/fr/free/nrw/commons/description/DescriptionEditActivityUnitTest.kt [hunk 3]
```diff
@@ -100,29 +100,29 @@ class DescriptionEditActivityUnitTest {
...
-    fun testOnSubmitButtonClicked() {
-        Shadows.shadowOf(Looper.getMainLooper()).idle()
-        val method: Method = DescriptionEditActivity::class.java.getDeclaredMethod(
-            "onSubmitButtonClicked", View::class.java
-        )
-        method.isAccessible = true
-        method.invoke(activity, null)
-        assertEquals(activity.isFinishing, true)
-    }
+//    @Test
```
Comment (RitikaPahwa4444) [inline app/src/test/kotlin/fr/free/nrw/commons/description/DescriptionEditActivityUnitTest.kt:103]: Are these tests no longer required?

**Review by RitikaPahwa4444 — CHANGES_REQUESTED** [review]

File: app/src/test/kotlin/fr/free/nrw/commons/description/DescriptionEditActivityUnitTest.kt (hunk 3 above)
Comment (shashankiitbhu) [reply app/src/test/kotlin/fr/free/nrw/commons/description/DescriptionEditActivityUnitTest.kt]: Sorry for this, I was going to add a To-Do and ask for instructions on this from mentors but then I forgot about it, anyways, fortunately ,I have solved it now.

**Comment by shashankiitbhu** [comment]
@RitikaPahwa4444 made the required changes, please review it, thanks

**Comment by RitikaPahwa4444** [comment]
Sorry for the delay! Thank you for making the suggested changes. I'll test it over the weekend and share my observations.

**Comment by shashankiitbhu** [comment]
@RitikaPahwa4444 Sure

**Comment by shashankiitbhu** [comment]
@RitikaPahwa4444 @nicolas-raoul Please Review and let me know if there is any change required.

**Comment by shashankiitbhu** [comment]
@RitikaPahwa4444 Just Pinging In case you missed it, Please Review and merge this PR, thanks

**Comment by RitikaPahwa4444** [comment]
Sorry, I was facing some issues with my emulators, had to reset everything. Tested, this does fix the crash :)

**Review by RitikaPahwa4444 — APPROVED** [review]

**Comment by shashankiitbhu** [comment]
@RitikaPahwa4444 @nicolas-raoul if these changes fix the issue then can you please Merge this, thanks

**Comment by nicolas-raoul** [comment]
Sorry would you mind rebasing from main?
I am getting this error:
```
> Task :app:compileProdDebugJavaWithJavac
The following annotation processors are not incremental: butterknife-compiler-10.1.0.jar (com.jakewharton:butterknife-compiler:10.1.0), dagger-android-processor-2.23.jar (com.google.dagger:dagger-android-processor:2.23), dagger-compiler-2.23.jar (com.google.dagger:dagger-compiler:2.23).
Make sure all annotation processors are incremental to improve your build speed.
/Users/nicolasraoul/src/commons-app/app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java:405: error: cannot find symbol
        progressBarEditDescription.setVisibility(GONE);
        ^
  symbol:   variable progressBarEditDescription
  location: class MediaDetailFragment
/Users/nicolasraoul/src/commons-app/app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java:406: error: cannot find symbol
        editDescription.setVisibility(VISIBLE);
        ^
  symbol:   variable editDescription
  location: class MediaDetailFragment
```

**Comment by shashankiitbhu** [comment]
@nicolas-raoul Done

**Comment by nicolas-raoul** [comment]
I am still getting this crash with commit c114c273bfec36b7d1ca0a1552d37c237fae2f69 when following the steps described in the bug description:

**Comment by shashankiitbhu** [comment]
@nicolas-raoul Fixed the Error, Please Review

**Comment by nicolas-raoul** [comment]
Would you mind fixing the conflict?

**Comment by shashankiitbhu** [comment]
@nicolas-raoul Fixed the merge conflicts , Please Review and Merge

**Review by nicolas-raoul — APPROVED** [review]
I confirm the fix for both "back" and "submit"

**Pull request merged**